service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureStorageAccessService.java

```diff
@@ -168,4 +224,27 @@ public AzureSasBundle createAzureStorageContainerSasToken(
             storageContainerResource.getStorageContainerName(),
             token));
   }
+
+  /**
+   * Returns an Azure container client suitable for interacting with a storage container and it's
+   * constiuent blobs.
+   *
+   * @param containerResource The WSM container resource the client will operate on
+   * @param storageAccountResource The parent storage account for the WSM container resource
+   * @return An Azure blob container client
+   */
+  public BlobContainerClient buildBlobContainerClient(
+      ControlledAzureStorageContainerResource containerResource,
+      ControlledAzureStorageResource storageAccountResource) {
```

Comment:
As we've talked about, we won't have a parent controlled storageAccountResource in the case of a landing zone storage account. It would be good to have a ticket for the subsequent work that will need to be done once TOAZ-264 is settled.

Reply:
https://broadworkbench.atlassian.net/browse/WOR-620